profiler/src/main/java/com/splunk/opentelemetry/profiler/allocation/exporter/PlainTextAllocationEventExporter.java

```diff
@@ -83,17 +81,14 @@ public void export(RecordedEvent event, AllocationEventSampler sampler, SpanCont
 
     Attributes attributes = builder.build();
 
-    LogDataBuilder logDataBuilder =
-        LogDataBuilder.create(resource, INSTRUMENTATION_SCOPE_INFO)
-            .setEpoch(time)
-            .setBody(body)
-            .setAttributes(attributes);
+    LogRecordBuilder logRecordBuilder =
+        logEmitter.logRecordBuilder().setEpoch(time).setBody(body).setAllAttributes(attributes);
 
     if (spanContext != null && spanContext.isValid()) {
-      logDataBuilder.setSpanContext(spanContext);
+      logRecordBuilder.setContext(Context.root().with(Span.wrap(spanContext)));
```

Comment:
What's up with this? Does this create a new context with a new span based on the existing context? I know you didn't write it, just trying to understand what/why...

Reply:
Yeah, pretty much. The `Span.wrap()` function creates just a shell/wrapper, not a fully fledged span though.

Unfortunately you can't just pass a `SpanContext` to a `LogRecordBuilder` directly, it has to be a `Context`.

Reply:
Ah, cool, thanks for clarifying `Context` vs. `SpanContext` (wasn't immediately obv to me, sorry). 👍🏻